Detectors/CPV/calib/testWorkflow/NoiseCalibratorSpec.h

```diff
@@ -0,0 +1,165 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Framework/Task.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/WorkflowSpec.h"
+#include "Framework/CCDBParamSpec.h"
+#include "CCDB/CcdbApi.h"
+#include "CCDB/CcdbObjectInfo.h"
+#include "DetectorsCalibration/Utils.h"
+#include "CPVCalibration/NoiseCalibrator.h"
+#include "DataFormatsCPV/Digit.h"
+#include "DataFormatsCPV/TriggerRecord.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace calibration
+{
+class CPVNoiseCalibratorSpec : public o2::framework::Task
+{
+ public:
+  //_________________________________________________________________
+  void init(o2::framework::InitContext& ic) final
+  {
+    uint64_t slotL = ic.options().get<uint64_t>("tf-per-slot");
+    uint64_t delay = ic.options().get<uint64_t>("max-delay");
+    uint64_t updateInterval = ic.options().get<uint64_t>("updateInterval");
+    bool updateAtTheEndOfRunOnly = ic.options().get<bool>("updateAtTheEndOfRunOnly");
+    mCalibrator = std::make_unique<o2::cpv::NoiseCalibrator>();
+    mCalibrator->setSlotLength(slotL);
+    mCalibrator->setMaxSlotsDelay(delay);
+    if (updateAtTheEndOfRunOnly) {
+      mCalibrator->setUpdateAtTheEndOfRunOnly();
+    }
+    mCalibrator->setCheckIntervalInfiniteSlot(updateInterval);
+    LOG(info) << "CPVNoiseCalibratorSpec initialized";
+    LOG(info) << "tf-per-slot = " << slotL;
+    LOG(info) << "max-delay = " << delay;
+    LOG(info) << "updateInterval = " << updateInterval;
+    LOG(info) << "updateAtTheEndOfRunOnly = " << updateAtTheEndOfRunOnly;
+  }
+  //_________________________________________________________________
+  void run(o2::framework::ProcessingContext& pc) final
+  {
+    auto tfcounter = o2::header::get<o2::framework::DataProcessingHeader*>(pc.inputs().get("digits").header)->startTime;
+    auto&& digits = pc.inputs().get<gsl::span<o2::cpv::Digit>>("digits");
+    auto&& trigrecs = pc.inputs().get<gsl::span<o2::cpv::TriggerRecord>>("trigrecs");
+
+    // read pedestal efficiencies, dead and high ped channels from pedestal run
+    // do it only once as they don't change during noise scan
+    if (!mCalibrator->mPedEfficiencies) {
+      const auto pedEffs = o2::framework::DataRefUtils::as<CCDBSerialized<std::vector<float>>>(pc.inputs().get("pedeffs"));
+      if (pedEffs) {
+        mCalibrator->mPedEfficiencies = new std::vector<float>(pedEffs->begin(), pedEffs->end());
```

Comment:
I see you are creating a local copy of the CCDB objects. The method I've suggested you did not work?

Reply:
Hi @shahor02 
Your method is working very fine, thank you! I use it in Raw2DigitsConverterSpec. 
In this noise calibration I decided to make local copy in very beginning of calibration only once and don't ask for this objects in next TFs as they will never change during noise scan. Is it ok?

Reply:
No problem to create them once, but the comment above about creating a vector with `new` and not deleting it afterwards.